scipy/stats/_rvs_sampling.py

```diff
@@ -168,3 +171,372 @@ def rvs_ratio_uniforms(pdf, umax, vmin, vmax, size=1, c=0, random_state=None):
+    ``H(p) == x``, where ``p`` is the array of percentiles corresponding
+    with the quantiles ``x``. Therefore, the spline approximates the inverse
+    of the distribution's CDF to machine precision at the percentiles ``p``,
+    but typically, the spline will not be as accurate at the midpoints between
+    the percentile points::
+
+        p_mid = (p[:-1] + p[1:])/2
+
+    so the mesh of quantiles is refined as needed to reduce the maximum
+    "u-error"::
+
+        u_error = np.max(np.abs(dist.cdf(H(p_mid)) - p_mid))
+
+    below the specified tolerance `tol`. Refinement stops when the required
+    tolerance is achieved or when the number of mesh intervals after the next
+    refinement could exceed the maximum allowed number `max_intervals`.
+
+    The approximation will not be accurate for the extreme tails of the
+    distribution (e.g. ``H(1e-17)``, ``H(1)``).
+
+    References
+    ----------
+    .. [1] Hörmann, Wolfgang, and Josef Leydold. "Continuous random variate
+           generation by fast numerical inversion." ACM Transactions on
+           Modeling and Computer Simulation (TOMACS) 13.4 (2003): 347-362.
+
+    Examples
+    --------
+    For some distributions, ``dist.ppf`` are ``dist.rvs`` are quite slow.
+
+    >>> import numpy as np
+    >>> from scipy.stats import genexpon
+    >>> from timeit import timeit
+    >>> time_once = lambda f: f"{timeit(f, number=1)*1000:.6} ms"
+    >>> dist = genexpon(9, 16, 3)  # freeze the distribution
+    >>> p = np.linspace(0.01, 0.99, 99)  # percentiles from 1% to 99%
+    >>> time_once(lambda: dist.ppf(p))
+    '154.565 ms'  # may vary
+
+    >>> time_once(lambda: dist.rvs(size=100))
+    '148.979 ms'  # may vary
+
+    The `FastNumericalInverse` has a method that approximates ``dist.ppf``.
+
+    >>> from scipy.stats import FastNumericalInverse
+    >>> fni = FastNumericalInverse(dist)
+    >>> np.allclose(fni.ppf(p), dist.ppf(p))
+    True
+
+    In some cases, it is faster to both generate the fast numerical inverse
+    and use it than to call ``dist.ppf``.
+
+    >>> def time_me():
+    ...     fni = FastNumericalInverse(dist)
+    ...     fni.ppf(p)
+    >>> time_once(time_me)
+    '11.9222 ms'  # may vary
+
+    After generating the fast numerical inverse, subsequent calls to its
+    methods are much faster.
+    >>> time_once(lambda: fni.ppf(p))
+    '0.0819 ms'  # may vary
+
+    The fast numerical inverse can also be used to generate random variates
+    using inverse transform sampling.
+
+    >>> time_once(lambda: fni.rvs(size=100))
+    '0.0911 ms'  # may vary
+
+    Depending on the implementation of the distribution's random sampling
+    method, the random variates generated may be nearly identical, given
+    the same random state.
+
+    >>> seed = 500072020
+    >>> rvs1 = dist.rvs(size=100, random_state=np.random.default_rng(seed))
+    >>> rvs2 = fni.rvs(size=100, random_state=np.random.default_rng(seed))
```

Comment:
There are 2 options here to not have the seed. Either deepcopy `rng = np.random.default_rng()` or store the generator state more explicitly:

```python
>>> state_rng = rng.__getstate__()
>>> rng.random(5)
# array([0.46605406, 0.57660729, 0.51196252, 0.38793054, 0.45671668])
>>> rng.__setstate__(state_rng)
>>>  rng.random(5)
# array([0.46605406, 0.57660729, 0.51196252, 0.38793054, 0.45671668])
```

Reply:
The example as-is more clearly demonstrates the point, IMO. In particular, it's an example of what _not_ to do, so I don't think there is any real danger of people copying this code straight out, which was the main concern for removing explicit `seed`s from other examples.

Reply:
mmm ok but then I would still add a comment `>>> seed = 500072020  # do not copy this value`

Reply:
Personally, I find those kinds of comments unpleasant when reading documentation and unfriendly to the reader. I would rather have a comment explaining the effect that the "bad" example is trying to achieve, like "create identical random streams to pass through each of the rvs() methods".

But if you'd insist on it, then I would avoid the `seed` and use `rng2 = copy.copy(rng)` with the above comment.

Reply:
I would do the `copy` then. I am afraid that letting one seed slide would make a future argument for accepting it in other examples 😅
Is that ok with you Matt?

Reply:
Well... honestly? I understand the importance of educating users about proper use of RNG. (For instance, as I was writing my thesis, I noticed that - if I remember correctly - Matlab suddenly decided to use some terrible RNG algorithm for code within `parfor` loops, and I took a lot of extra care to get around that.) But if that's the intent, I think we should do that explicitly rather than eliminating all use of seeds from the documentation.

Reply:
Seems like 2 to 1 here 😅  You can keep it like this it's fine. Hopefully the weird value will make a user not to use it. Let's just be sure to motivate such decision in the future, that's all I am asking 😉

Reply:
Ok, I'll add a note.